aviatrix/resource_aviatrix_spoke_gateway.go

```diff
@@ -1270,6 +1352,10 @@ func resourceAviatrixSpokeGatewayUpdate(d *schema.ResourceData, meta interface{}
 			newHaGwEnabled = true
 		} else if deleteHaGw {
 			//Ha configuration has been deleted
+			if d.Get("ha_gw_size").(string) != "" {
+				return fmt.Errorf("\"ha_gw_size\" must be empty if spoke HA gateway is deleted")
+			}
+
```

Comment:
Should we also validate the other `ha_*` attributes? (ha_subnet, ha_eip, ha_zone)

Reply:
There's already a cidr check for `ha_subnet`.
We can add ip address validation for `eip` and `ha_eip`.
For `ha_zone`, I'm not sure. We have a validation function for azure, but azure's az format is simple. For other providers, we'll have to maintain a list of az. I think we can add this later.